packages/sync-rules/src/SqlDataQuery.ts

```diff
@@ -13,14 +13,23 @@ import { TableQuerySchema } from './TableQuerySchema.js';
 import { EvaluationResult, ParameterMatchClause, QuerySchema, SqliteRow } from './types.js';
 import { getBucketId, isSelectStatement } from './utils.js';
 
-export class SqlDataQuery extends BaseSqlDataQuery {
-  filter?: ParameterMatchClause;
+export interface SqlDataQueryOptions extends BaseSqlDataQueryOptions {
+  filter: ParameterMatchClause;
+}
 
-  static fromSql(descriptor_name: string, bucket_parameters: string[], sql: string, options: SyncRulesOptions) {
+export class SqlDataQuery extends BaseSqlDataQuery {
+  static fromSql(
+    descriptorName: string,
+    bucketParameters: string[],
+    sql: string,
+    options: SyncRulesOptions,
+    ruleId?: string
```

Comment:
I'm not sure I fully understand the purpose of `ruleId` from a quick look, but it looks like this value is always set (and only ignored for tests). So I wonder if we should make it required and then set it to a bogus value in tests directly calling `SqlDataQuery.fromSql`.

Reply:
Good point. I looked at this again and realized for data queries it was only used in 2 tests, and nowhere outside tests. I removed it rather.